proto/proto/store/shared.proto

```diff
@@ -0,0 +1,278 @@
+// ================================================================================================
+
+// Represents the status of the store.
+message StoreStatus {
+    // The store's running version.
+    string version = 1;
+
+    // The store's status.
+    string status = 2;
+
+    // Number of the latest block in the chain.
+    fixed32 chain_tip = 3;
+}
+
+// GET ACCOUNT STATE DELTA
+// ================================================================================================
+
+// Returns delta of the account states in the range from `from_block_num` (exclusive) to
+// `to_block_num` (inclusive).
+message GetAccountStateDeltaRequest {
+    // ID of the account for which the delta is requested.
+    account.AccountId account_id = 1;
+    // Block number from which the delta is requested (exclusive).
+    fixed32 from_block_num = 2;
+    // Block number up to which the delta is requested (inclusive).
+    fixed32 to_block_num = 3;
+}
+
+// Represents the result of getting account state delta.
+message AccountStateDelta {
+    // The calculated account delta encoded using [winter_utils::Serializable] implementation
+    // for [miden_objects::account::delta::AccountDelta].
+    optional bytes delta = 1;
+}
+
+// CHECK NULLIFIERS BY PREFIX
+// ================================================================================================
+
+// Returns a list of nullifiers that match the specified prefixes and are recorded in the node.
+message CheckNullifiersByPrefixRequest {
+    // Number of bits used for nullifier prefix. Currently the only supported value is 16.
+    uint32 prefix_len = 1;
+    // List of nullifiers to check. Each nullifier is specified by its prefix with length equal
+    // to `prefix_len`.
+    repeated uint32 nullifiers = 2;
+    // Block number from which the nullifiers are requested (inclusive).
+    fixed32 block_num = 3;
+}
+
+// Represents the result of checking nullifiers by prefix.
+message CheckNullifiersByPrefixResponse {
+    // Represents a single nullifier update.
+    message NullifierUpdate {
+        // Nullifier ID.
+        primitives.Digest nullifier = 1;
+
+        // Block number.
+        fixed32 block_num = 2;
+    }
+    // List of nullifiers matching the prefixes specified in the request.
+    repeated NullifierUpdate nullifiers = 1;
+}
+
+// CHECK NULLIFIERS
+// ================================================================================================
+
+// List of nullifiers to return proofs for.
+message NullifierList {
+    // List of nullifiers to return proofs for.
+    repeated primitives.Digest nullifiers = 1;
+}
+
+// Represents the result of checking nullifiers.
+message CheckNullifiersResponse {
+    // Each requested nullifier has its corresponding nullifier proof at the same position.
+    repeated primitives.SmtOpening proofs = 1;
+}
+
+// GET BLOCK HEADER BY NUMBER
+// ================================================================================================
```

Comment:
Related to the previous comment: I think these may be the only shared messages between the stores sub-services.